zipline/utils/input_validation.py

```diff
@@ -130,6 +133,19 @@ def ensure_timezone(func, argname, arg):
     )
 
 
+def expect_non_empty(func, argname, arg):
```

Comment:
I thought the more important pattern was between `expect` and `ensure`. I could do that but I still would want a named version for `non_empty` which is a more common case imho